src/parser/parser.cc

```diff
@@ -257,25 +325,27 @@ Parser::parse_function_declaration() {
     RETURN_IF_ERROR(get_token());
     RETURN_OR_MOVE(type, parse_type());
   }
-  auto body_location = scoped_location();
+
   if (current_token().type() == TokenType::ASSIGN) {
     // fun my_fun() = 3;
     RETURN_IF_ERROR(get_token());
-    RETURN_OR_MOVE(std::unique_ptr<ast::Value> body, parse_value());
-    EXPECT_TOKEN(TokenType::SEMICOLON, "Expected `;' after function body");
+    RETURN_OR_MOVE(auto value, parse_value());
```

Comment:
Could be something else, what about

fun main() = if (condition) value else value ?